src/main/java/mate/academy/spring/dao/RoleDao.java

```diff
@@ -0,0 +1,9 @@
+package mate.academy.spring.dao;
+
+import mate.academy.spring.model.Role;
+
+public interface RoleDao {
+    Role add(Role role);
+
+    Role getByName(String roleName);

```

Comment:
Be careful in getByName() method on dao layer! You should pass enum to query (not String), otherwise you won’t be able to get Role by name.
```suggestion
   Optional<Role> getByName(Role.RoleName roleName);
```